btm/src/main/java/bitronix/tm/resource/ResourceRegistrar.java

```diff
@@ -47,24 +49,36 @@
 
     private final static Logger log = LoggerFactory.getLogger(ResourceRegistrar.class);
 
-    private static final Lock registrationLock = new ReentrantLock();
-    private static final ConcurrentMap<String, XAResourceProducer> resources = new ConcurrentHashMap<String, XAResourceProducer>();
+    private static ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
```

Comment:
rename to `registrationReadWriteLock`/`registrationWriteLock`/`registrationReadLock` for consistency with the previous name